diff --git a/dist/listeners/command-handler/CoreMessageParser.js b/dist/listeners/command-handler/CoreMessageParser.js
index 77085e6236e54886704803e8111278c4977780df..a76c155d757cc0e2890529ad6ad322246ce328ca 100644
--- a/dist/listeners/command-handler/CoreMessageParser.js
+++ b/dist/listeners/command-handler/CoreMessageParser.js
@@ -43,13 +43,16 @@ class CoreListener extends Listener_1.Listener {
     async canRunInChannel(message) {
         if ((0, discord_js_utilities_1.isDMChannel)(message.channel))
             return true;
-        const me = message.guild.me ?? (message.client.id ? await message.guild.members.fetch(message.client.id) : null);
+        const me = message.guild.members.me ?? (message.client.id ? await message.guild.members.fetch(message.client.id) : null);
         if (!me)
             return false;
         if (me.isCommunicationDisabled())
             return false;
         const channel = message.channel;
-        return channel.permissionsFor(me).has(this.requiredPermissions, false);
+        const permissionsFor = channel.permissionsFor(me);
+        if (!permissionsFor) return false;
+
+        return permissionsFor.has(this.requiredPermissions, false);
     }
     getMentionPrefix(message) {
         if (this.container.client.disableMentionPrefix)
@@ -59,7 +62,7 @@ class CoreListener extends Listener_1.Listener {
             return null;
         // Calculate the offset and the ID that is being provided
         const [offset, id] = message.content[2] === '&'
-            ? [3, message.guild?.roles.botRoleFor(message.guild.me)?.id]
+            ? [3, message.guild?.roles.botRoleFor(message.guild.members.me)?.id]
             : [message.content[2] === '!' ? 3 : 2, this.container.client.id];
         if (!id)
             return null;
