diff --git a/dist/index.js b/dist/index.js
index b9256ddb47c0bee60891aec094728daf7a806cec..02eb56e8ce49efc3ad84702e598a6e0be22290e3 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1323,7 +1323,16 @@ function canJoinVoiceChannel(channel) {
 }
 __name(canJoinVoiceChannel, "canJoinVoiceChannel");
 function canDoUtility(channel, permissionsToPass) {
-  return isGuildBasedChannel(channel) ? channel.permissionsFor(channel.guild.me).has(permissionsToPass) : true;
+  if (!isGuildBasedChannel(channel)) {
+    return true;
+  }
+  const { me } = channel.guild;
+  if (!me)
+    return false;
+  const permissionsFor = channel.permissionsFor(me);
+  if (!permissionsFor)
+    return false;
+  return permissionsFor.has(permissionsToPass);
 }
 __name(canDoUtility, "canDoUtility");
 // Annotate the CommonJS export names for ESM import in node:
