diff --git a/src/client/actions/ChannelUpdate.js b/src/client/actions/ChannelUpdate.js
index 34d1a86a4c0d59317b7d7be69b42cff4fb71e1c7..3695e0b0a2073365378b6d3be9cadff5e718a350 100644
--- a/src/client/actions/ChannelUpdate.js
+++ b/src/client/actions/ChannelUpdate.js
@@ -7,14 +7,23 @@ const { ChannelTypes } = require('../../util/Constants');
 class ChannelUpdateAction extends Action {
   handle(data) {
     const client = this.client;
-
     let channel = client.channels.cache.get(data.id);
+
     if (channel) {
       const old = channel._update(data);
 
       if (ChannelTypes[channel.type] !== data.type) {
         const newChannel = Channel.create(this.client, data, channel.guild);
-        for (const [id, message] of channel.messages.cache) newChannel.messages.cache.set(id, message);
+
+        if (!newChannel) {
+          this.client.channels.cache.delete(channel.id);
+          return {};
+        }
+
+        if (channel.isText() && newChannel.isText()) {
+          for (const [id, message] of channel.messages.cache) newChannel.messages.cache.set(id, message);
+        }
+
         channel = newChannel;
         this.client.channels.cache.set(channel.id, channel);
       }
